core/src/scene/sceneLayer.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <unordered_map>
+
+#include "filters.h"
+#include "style/styleParamMap.h"
+
+namespace Tangram {
+
+    class SceneLayer {
+        std::vector<SceneLayer*> m_subLayers;
```

Comment:
I can do vector of shared_ptr. 

Why not vector of values: in applyLayerFiltering I am using a local vector to store all the layers for BFS traversal. If I do by value I will be make a copy for every layer, which I can avoid.